### ISSUE TITLE
How to fetch data and don't update the results after mutation

### ISSUE THREAD

**Issue by thiagonzalez** [aribouius/jsonapi-react#48]
Hello guys,

I have a page that needs to verify if any applications were created by the user to show or not a content. If the user has one or more applications, I redirect him to the applications page. If the user doesn't have application, a tutorial will start to create a first application.

When the user creates an application using the tutorial, the lib refetches data and redirect the user, because now he has an application, but the tutorial still has some additional steps after this creation.

Can I prevent refetching applications in this situation? You can find the snippet below:

```
export default function Dashboard() {
  const { data, isLoading } = useQuery(['apps']);

  if (isLoading) return <Loading />;

  if (data.length > 0) {
    /* function to redirect users here */
    return <Loading />;
  }

  return <Tutorial />;
}
```

**Comment by aribouius** [comment]
Hi @thiagonzalez,

Yes this is just requires instructing the mutation hook to _not_ attempt to perform any [cache invalidation](https://github.com/aribouius/jsonapi-react#invalidation).

```javascript
const [mutation] = useMutation('apps', {
  invalidate: false
})
```

Let me know if that works for you!

**Comment by thiagonzalez** [comment]
hey @aribouius, thanks for your response, but the param `invalidate` doesn't accept boolean, only string or array of strings:

![image](https://user-images.githubusercontent.com/967571/146441705-ec62cdfa-36f2-4251-81dd-d95a1fe3f076.png)

**Comment by aribouius** [comment]
@thiagonzalez ah sorry about that, I'm not a TS user so I hadn't noticed this before.  The `invalidate` option is intended to allow a _falsey_ value (e.g. `false` and `null`), but the [types config](https://github.com/aribouius/jsonapi-react/blob/master/types/index.d.ts#L92) wasn't wired up correctly.

Would you be comfortable submitting a PR that updates the types?

**Comment by thiagonzalez** [comment]
@aribouius thanks for the opportunity to contribute to your project!
Please, have a look at my [PR](https://github.com/aribouius/jsonapi-react/pull/49). I added an explanation on the README and also added `boolean` as one of the types for `invalidate`.

**Comment by aribouius** [comment]
Thanks for your contribution @thiagonzalez, and apologies for the delay in getting your PR merged, I didn't have much time to work on this over the holidays.

Your update is now available on version `0.0.25`.

**Comment by thiagonzalez** [comment]
No worries at all, @aribouius. Thank you so much for your incredible work here!
Happy new year! :)
